src/test/users-query.test.ts

```diff
@@ -0,0 +1,145 @@
+import { expect } from 'chai';
+import * as faker from 'faker';
+import  requests from 'supertest';
+import { getRepository, Repository } from 'typeorm';
+import { User } from '../entities/user';
+import { hashPassword } from '../hash-password';
+import * as jwt from 'jsonwebtoken';
+import {AUTHEN_ERROR} from '../errors';
+import {TEN_MINUTES, APP_SECRET} from '../consts';
+
+describe('Users query tests', () => {
+
+  let agent;
+  let repository: Repository<User>;
+
+  const usersQuery = (token, variable?: { limit?, offset?}) => {
+
+    const query = `
+    query ($limit: Int, $offset: Int ) {
+      users (limit: $limit, offset: $offset ) {
+        users {
+          id 
+          name 
+          email 
+          birthDate 
+          cpf
+        }
+        total
+        info {
+          before
+          after
+        }
+      }
+    }
+    `
+    return agent.post("/").send({ query: query, variables: variable }).set('Authorization', token)
+  }
+
+  before(async () => {
+
+    agent = requests('http://localhost:4000');
+    
+    repository = getRepository(User)
+
+    const newusers = Array.from({ length: 55 }).map(() => {
+        const newUser: User = new User();
+        newUser.name = faker.name.findName();
+        newUser.email = faker.internet.email();
+        newUser.birthDate = faker.date.past();
+        newUser.cpf = faker.random.number();
+        newUser.password = hashPassword('password1');
+        return newUser;
+      })
+
+    await repository.save(newusers)
+
+  })
+
+  after( async () => {
+    await repository.clear()
+  })
+
+  it('should return the first ten users', async function() {
+      var token = jwt.sign({ userId: 1 }, APP_SECRET, { expiresIn: TEN_MINUTES }); 
```

Comment:
Como essa declaração de `token` se repete em quase todos os testes, talvez seja uma boa colocar ela no `beforeEach` e, quando for o caso, apenas sobrescrevê-lo.